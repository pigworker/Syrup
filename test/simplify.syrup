not(<Bit>) -> <Bit>
not(X) = nand(X,X)

f(<Bit>) -> <Bit>
f(X) = not(not(not(not(not(X)))))

simplify f

g(<Bit>) -> <Bit>, <Bit>, <Bit>, <Bit>
g(X) = X,not(Y),Z,E where
  Y = not(not(Z))
  Z = not(E)
  E = not(X)

simplify g

and(<Bit>, <Bit>) -> <Bit>
and(X, Y) = not (nand (not (not(X)), Y))

simplify and

h(<Bit>, <Bit>) -> <Bit>
h(X,Y) = not (and(X, not(not(Y))))

simplify h

or(<Bit>, <Bit>) -> <Bit>
or(X, Y) = nand(not(X), not(Y))

i(<Bit>, <Bit>) -> <Bit>
i(X,Y) = not(or(not(X), Y))

simplify i

j(<Bit>, <Bit>) -> <Bit>
j(X,Y) = or(not(X), not(Y))

simplify j

k(<Bit>, <Bit>) -> <Bit>
k(X,Y) = or(or(not(X), not(Y)), not(X))

simplify k