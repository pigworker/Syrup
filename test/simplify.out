Circuit not is defined.

Circuit f is defined.

Simplification of f:
f(<Bit>) -> <Bit>
f(X) = !X

Circuit g is defined.

Simplification of g:
g(<Bit>) -> <Bit>, <Bit>, <Bit>, <Bit>
g(X) = X, !X, Z, E where
  E = !X
  Z = X

Circuit and is defined.

Simplification of and:
<Bit> & <Bit> -> <Bit>
X & Y = !nand(X, Y)

Circuit h is defined.

Simplification of h:
h(<Bit>, <Bit>) -> <Bit>
h(X, Y) = nand(X, Y)

Circuit or is defined.

Circuit i is defined.

Simplification of i:
i(<Bit>, <Bit>) -> <Bit>
i(X, Y) = X & !Y

Circuit j is defined.

Simplification of j:
j(<Bit>, <Bit>) -> <Bit>
j(X, Y) = nand(X, Y)

Circuit k is defined.

Simplification of k:
k(<Bit>, <Bit>) -> <Bit>
k(X, Y) = nand(X & Y, X)

Circuit l is defined.

Simplification of l:
l(<Bit>, <Bit>) -> <Bit>
l(X, Y) = X & Y

Circuit m is defined.

Simplification of m:
m(<Bit>, <Bit>) -> <Bit>
m(X, Y) = nand(X, Y)
